apps/emqx_management/src/emqx_mgmt_api_clients.erl

```diff
@@ -1738,16 +1738,15 @@ format_channel_info(undefined, {ClientId, PSInfo0 = #{}}, _Opts) ->
 
 format_persistent_session_info(
     _ClientId,
-    #{
-        metadata := #{offline_info := #{chan_info := ChanInfo, stats := Stats} = OfflineInfo} =
-            Metadata
-    } =
-        PSInfo
+    #{metadata := Metadata} = PSInfo
 ) ->
+    %% FIXME: this function should not access internal representation
+    %% of DS state directly
+    #{offline_info := #{chan_info := ChanInfo, stats := Stats} = OfflineInfo} = Metadata,
```

Comment:
Without this in the function clause, the next clause won't be triggered if `offline_info` is missing for whatever reason.